packages/core/src/node/messaging/messaging-contribution.ts

```diff
@@ -107,7 +109,26 @@ export class MessagingContribution implements BackendApplicationContribution, Me
     }
 
     protected handleChannels(socket: Socket): void {
-        const socketChannel = new WebSocketChannel(this.toIWebSocket(socket));
+        const reconnectionKey = new URL(socket.request.url!, 'http://localhost/').searchParams.get(PersistentWebSocket.ReconnectionKey) || '';
+        const toClose = new Array<PersistentWebSocket>();
+        this.persistentConnections.forEach((value, key) => {
+            if (key !== reconnectionKey && !value.underlyingSocketConnected) {
+                toClose.push(value);
+            }
+        });
+        // close other non-connected connections when there is a connection in
+        toClose.forEach(value => value.fireClose());
```

Comment:
I don't understand this logic:

1. We handle 1 channel reconnection with some key
2. For all opened connections that don't match that key we fire the close event on those??
3. Looking at the implementation for `fireClose` it seems to shortcut `PersistentWebSocket`'s internal state, as in it doesn't do anything but just fires the event?

Reply:
Here's what I understand:

1. I have 3 Theia tabs opened to the same backend on different workspaces
2. I lose connection to the backend in all 3 tabs
3. One tab reconnects before the others
4. The 2 other slower tabs get their connection closed?

Reply:
we can change the behavior, to wait for every connection for the same timeout